neutron/convert.cpp

```diff
@@ -128,7 +134,8 @@ Dataset tofToEnergy(Dataset &&d) {
 
 Dataset energyToTof(Dataset &&d) {
   // 1. Compute conversion factor
-  const auto conversionFactor = tofToEnergyConversionFactor(d);
+  const auto conversionFactor =
+      astype(tofToEnergyConversionFactor(d), d.coords()[Dim::Energy].dtype());
```

Comment:
Same comment as above: I think we cannot convert the dtype of `conversionFactor`, but have to do it individually for each coord (global dense coord, and sparse coords).